h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/XGBoostUtils.java

```diff
@@ -295,12 +286,12 @@ public static DMatrix convertChunksToDMatrix(DataInfo di,
                 // truly sparse matrix - no categoricals
                 // collect all nonzeros column by column (in parallel), then stitch together into final data structures
                 if (csc) {
-                    trainMat = csc(chunks, weight, nRows, di, resp, weights);
+                    trainMat = csc(chunks, -1, nRows, di, resp, weights);
```

Comment:
This method is currently preds-only, but I would avoid these changes. Is narrowing this method's scope what we really want ? If yes, when does the method have correct name ? Leaving the method with more general scope is one way to go.

Otherwise, we can simply discard `float[] weights = null;` on line 277 and pass null explicitly on each line, as this directly shows that weights are always null.

Reply:
this change does nothing to the method scope, given functionality was never used
cleaned up even furhter

Reply:
By scope, I meant how general the algorithm represented by the method is.

I am not a code junkie. I agree this functionality was unused and can be added on demand in future easily. No big deal. The decision is yours. I'm ok with it as long as it's correct, and it is.